app/controllers/cartitems_controller.rb

```diff
@@ -0,0 +1,78 @@
+class CartitemsController < ApplicationController
+  before_action :find_cartitem, only: [:show, :update, :destroy]
+  before_action :cartitem_params, only: [:update, :create]
+
+  def create
+    if session[:cart_id]
+      @cart = Cart.find_by(id: session[:cart_id])
+      @cartitem = Cartitem.new(cartitem_params)
```

Comment:
This method is not very DRY. I would try and separate the cart creation logic out, something like this:

```ruby
def create
  @cart = Cart.find_by(id: session[:cart_id])
  unless @cart
    # Create the cart or return an error
  end

  # If we get to this point then we definitely have a cart
  # stored in @cart

  @cartitem = @cart.cartitems.new(...)
  # Save the cartitem or return an error...
end
```